Sources/GRPC/ConnectionManager.swift

```diff
@@ -176,38 +176,55 @@ internal final class ConnectionManager {
     }
   }
 
+  /// The last 'external' state we are in, a subset of the internal state.
+  private var externalState: ConnectivityState = .idle
+
+  /// Update the external state, potentially notifying a delegate about the change.
+  private func updateExternalState(to nextState: ConnectivityState) {
+    if self.externalState != nextState {
+      self.connectivityDelegate?.connectionStateDidChange(
+        self, from: self.externalState, to: nextState
+      )
+      self.externalState = nextState
```

Comment:
Is this order of operations correct? Generally my advice is to update our internal state _first_ and then make arbitrary outcalls second. Should we be following that advice here?

Reply:
Oof yes, good catch.